monitor/src/file_utils.rs

```diff
@@ -0,0 +1,103 @@
+use crate::constants::{ACTIONS_DIR, DIRECTORY_CREATION_ERROR, EVENT, FILE_CREATION_ERROR, GITHUB_TOKEN, REPO_NAME, REPO_OWNER};
+use actix_multipart::Multipart;
+use actix_web::Error;
+use futures::TryStreamExt;
+use serde::Deserialize;
+use std::fs::{create_dir_all, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+#[derive(Deserialize)]
+pub struct NewConfig {
+    pub(crate) event: String,
+    pub(crate) repo_owner: String,
+    pub(crate) repo_name: String,
+    pub(crate) github_token: String
+}
+
+pub struct MultipartResult {
+    pub(crate) new_config: NewConfig,
+    pub(crate) actions_file_path: Option<String>
+}
+
+// Helper function to create a directory if it doesn't exist
+fn create_directory_if_not_exists(path: &Path) -> actix_web::Result<(), Error> {
```

Comment:
You are using this function in  a http handler. 2 solutions: 
- You can return classic error and then catch them upward 
- You can return an HttpResponse type with for example:
```rs
HttpResponse::BadRequest(...)
```